### PR TITLE
Enable TLS native root toggling at runtime

### PR DESCRIPTION
**Pull request by charliermarsh**
## Summary

It turns out that on macOS, reading the native certificates can add hundreds of milliseconds to client initialization. This PR makes `--native-tls` a command-line flag, to toggle (at runtime) the choice of the `webpki` roots or the native system roots.

You can't accomplish this kind of configuration with the `reqwest` builder API, so instead, I pulled out the heart of that logic from the crate (https://github.com/seanmonstar/reqwest/blob/e3192638518d577759dd89da489175b8f992b12f/src/async_impl/client.rs#L498), and modified it to allow toggling a choice of root.

Note that there's an open PR for this in reqwest (https://github.com/seanmonstar/reqwest/pull/1848), along with an issue (https://github.com/seanmonstar/reqwest/issues/1843), which I may ping, but it's been around for a while and I believe reqwest is focused on its next major release.

Closes https://github.com/astral-sh/uv/issues/2346.

### CODE REVIEW

File: crates/uv/src/main.rs [hunk 1]
```diff
@@ -88,6 +88,13 @@ struct Cli {
     )]
     color: ColorChoice,
 
+    /// Whether to load TLS certificates from the platform's native certificate store.
+    ///
+    /// By default, `uv` loads certificates from the bundled `webpki-roots` crate, which contains
+    /// Mozilla's root certificates.
```
Comment (zanieb) [inline crates/uv/src/main.rs:94]: Should we say a little more about why the system store would be used?

File: README.md [hunk 2]
```diff
@@ -423,7 +423,7 @@ In addition, uv respects the following environment variables:
 
 uv supports custom CA certificates (such as those needed by corporate proxies) by utilizing the
 system's trust store. To ensure this works out of the box, ensure your certificates are added to the
-system's trust store.
+system's trust store, and run uv with the `--native-tls` command-line Flag.
```
Comment (zanieb) [inline README.md:426]: I guess I would rephrase since this isn't quite "out of the box" anymore?

Comment (zanieb) [reply README.md:426]: I'd say... "By default, we use Mozilla's...." as you do in the docstring as well

**Review by zanieb — APPROVED** [review]
Great work!

**Comment by zanieb** [comment]
Technically this is a breaking change, should we create a label for that and highlight accordingly?

**Comment by samypr100** [comment]
@charliermarsh Is it worth checking if `SSL_CERT_FILE` is set and make it behave as-if `--native-tls` was used?
I think that would keep this change backwards compatible with users leveraging `SSL_CERT_FILE`

**Comment by charliermarsh** [comment]
We can support `SSL_CERT_FILE` even without `--native-tls`, I think. I'm happy to add that. It would effectively be this PR: https://github.com/astral-sh/uv/pull/2351.
